### ISSUE TITLE
bun start crash - TypeError: undefined is not an object at terminate()

### ISSUE THREAD

**Issue by zardazare** [oven-sh/bun#11972]
### What version of Bun is running?

1.1.14+7c27f3f9b

### What platform is your computer?

Darwin 23.4.0 x86_64 i386

### What steps can reproduce the bug?

`bun start`

Note: using webpack-dev-server: "3.11.1"

### What is the expected behavior?

Running application on localhost port after it is compiled successfully.

### What do you see instead?

```
505 |       this.#enquedMessages.push([data, opts?.compress, cb]), this.#bufferedAmount += data.length;
506 |   }
507 |   close(code, reason) {
508 |     if (this.#state === 1)
509 |       this.#state = 2, this.#ws.close(code, reason);
510 |   terminate() {
                 ^
TypeError: undefined is not an object
      at terminate (ws:510:12)
      at forEach (:1:11)
      at project_path/node_modules/webpack-dev-server/lib/servers/WebsocketServer.js:34:29

Bun v1.1.14 (macOS x64)
error: script "start" exited with code 1
```

**Comment by daytona63** [comment]
I am facing the same issue here on my graphql web socket server

**Comment by daytona63** [comment]
it seems like the issue comes up when I apply keepAlive in the `graphql-ws` server, and when they internally call `socket.terminate();`, it crashes the server.

```js
   const pingInterval = keepAlive > 0 && isFinite(keepAlive)
            ? setInterval(() => {
                // ping pong on open sockets only
                if (socket.readyState === socket.OPEN) {
                    // terminate the connection after pong wait has passed because the client is idle
                    pongWait = setTimeout(() => {
                        socket.terminate();
                    }, keepAlive);
                    // listen for client's pong and stop socket termination
                    socket.once('pong', () => {
                        if (pongWait) {
                            clearTimeout(pongWait);
                            pongWait = null;
                        }
                    });
                    socket.ping();
                }
            }, keepAlive)
            : null;
```


Error

```
511 |       this.#enquedMessages.push([data, opts?.compress, cb]), this.#bufferedAmount += data.length;
512 |   }
513 |   close(code, reason) {
514 |     if (this.#state === 1)
515 |       this.#state = 2, this.#ws.close(code, reason);
516 |   terminate() {
                 ^
TypeError: undefined is not an object
      at terminate (ws:516:12)
      at /Users/adelak/Projects/yalliving/node_modules/.pnpm/graphql-ws@5.16.0_graphql@16.9.0/node_modules/graphql-ws/lib/use/ws.mjs:57:32
```

**Comment by garysb** [comment]
I have the same issue, also with graphql-ws but I only get the error when running on a Kubernetes cluster and can not replicate on our devcontainer.

```
511 |       this.#enquedMessages.push([data, opts?.compress, cb]), this.#bufferedAmount += data.length;
512 |   }
513 |   close(code, reason) {
514 |     if (this.#state === 1)
515 |       this.#state = 2, this.#ws.close(code, reason);
516 |   terminate() {
                 ^
TypeError: undefined is not an object
      at terminate (ws:516:12)
      at /app/node_modules/@nestjs/graphql/node_modules/graphql-ws/lib/use/ws.js:60:32

Bun v1.1.22-canary.96+df33f2b2a (Linux arm64)
error: script "backend:api:start" exited with code 1
```

**Comment by daytona63** [comment]
> I have the same issue, also with graphql-ws but I only get the error when running on a Kubernetes cluster and can not replicate on our devcontainer.
> 
> ```
> 511 |       this.#enquedMessages.push([data, opts?.compress, cb]), this.#bufferedAmount += data.length;
> 512 |   }
> 513 |   close(code, reason) {
> 514 |     if (this.#state === 1)
> 515 |       this.#state = 2, this.#ws.close(code, reason);
> 516 |   terminate() {
>                  ^
> TypeError: undefined is not an object
>       at terminate (ws:516:12)
>       at /app/node_modules/@nestjs/graphql/node_modules/graphql-ws/lib/use/ws.js:60:32
> 
> Bun v1.1.22-canary.96+df33f2b2a (Linux arm64)
> error: script "backend:api:start" exited with code 1
> ```

@garysb i managed to get around it by setting keepAlive to 0

```
const keepAlive = 0;
...code here
return useServer(serverOptions, wsServer, keepAlive);
```

**Comment by UltimateGG** [comment]
Same issue with trpc websockets and keep alive on Bun 1.1.26
```
513 |   close(code, reason) {
514 |     if (this.#state === 1)
515 |       this.#state = 2, this.#ws.close(code, reason);
516 |   terminate() {
                 ^
TypeError: undefined is not an object
      at terminate (ws:516:12)
```

**Comment by marekhanzlik** [comment]
Bun `v1.1.36` still have this issue. But `bun upgrade --canary` (`v1.1.37`) fixed the problem for me. So either temporarily switch to canary, or wait for `v1.1.37`

**Comment by aptash** [comment]
I can also confirm the fix in canary (v1.1.37)
